packages/curve-ui-kit/src/features/connect-wallet/lib/wagmi/useWagmi.tsx

```diff
@@ -0,0 +1,54 @@
+import { Eip1193Provider } from 'ethers'
+import { useCallback, useState } from 'react'
+import { useAccount, useChainId, useClient, useConnect, useConnectors, useDisconnect, useEnsName } from 'wagmi'
+import type { Wallet } from '../types'
+import { clientToProvider } from './adapter'
+import { supportedWallets } from './wallets'
+
+export const useWagmi = () => {
+  const [label, setLabel] = useState('')
```

Comment:
useState is not appropriate when the hook is called multiple times.
What's wrong with
https://github.com/curvefi/curve-frontend/blob/66d310555145da413aedbc1259c9b867441a58a3/packages/curve-ui-kit/src/hooks/useLocalStorage.ts#L36

Reply:
I'm not sure if storing labels is required at all to be honest. Seems like it's a remainder from onboard? If you're intending to use labels for auto reconnect it's not needed, given that wagmi connectors already support that natively.

Reply:
I initially put the connector in the `WalletType` but then I couldn't retrieve the connector name anymore (dont remember why I needed that lol)